test/e2e/Tests/Tests/SuspendResumeTests.cs

```diff
@@ -0,0 +1,197 @@
+
+            using HttpResponseMessage suspendResponse = await HttpHelpers.InvokeHttpTrigger("SuspendInstance", $"?instanceId={instanceId}");
+            await AssertRequestSucceedsAsync(suspendResponse);
+
+            Thread.Sleep(1000);
+
+            using HttpResponseMessage resumeResponse = await HttpHelpers.InvokeHttpTrigger("SuspendInstance", $"?instanceId={instanceId}");
+            await AssertRequestFailsAsync(resumeResponse);
+
+            // Give some time for Core Tools to write logs out
+            Thread.Sleep(500);
+
+            Assert.Contains(_fixture.TestLogs.CoreToolsLogs, x => x.Contains("Cannot suspend orchestration instance in the Suspended state.") &&
+                                                                  x.Contains(instanceId));
+        }
+        finally
+        {
+            await TryTerminateInstanceAsync(instanceId);
+        }
+    }
+
+
+    [Fact]
+    public async Task ResumeRunningOrchestration_ShouldFail()
+    {
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("LongOrchestrator_HttpStart", "");
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string instanceId = await DurableHelpers.ParseInstanceIdAsync(response);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+
+        Thread.Sleep(1000);
+        try
+        {
+            var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+            Assert.Equal("Running", orchestrationDetails.RuntimeStatus);
+
+            Thread.Sleep(1000);
+
+            using HttpResponseMessage resumeResponse = await HttpHelpers.InvokeHttpTrigger("ResumeInstance", $"?instanceId={instanceId}");
+            await AssertRequestFailsAsync(resumeResponse);
+
+            // Give some time for Core Tools to write logs out
+            Thread.Sleep(500);
+
+            Assert.Contains(_fixture.TestLogs.CoreToolsLogs, x => x.Contains("Cannot resume orchestration instance in the Running state.") &&
+                                                                  x.Contains(instanceId));
+        }
+        finally
+        {
+            await TryTerminateInstanceAsync(instanceId);
+        }
+    }
+
+
+    [Fact]
+    public async Task SuspendResumeCompletedOrchestration_ShouldFail()
+    {
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("HelloCities_HttpStart", "");
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string instanceId = await DurableHelpers.ParseInstanceIdAsync(response);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+
+        Thread.Sleep(1000);
+        try
+        {
+            using HttpResponseMessage suspendResponse = await HttpHelpers.InvokeHttpTrigger("SuspendInstance", $"?instanceId={instanceId}");
+            await AssertRequestFailsAsync(suspendResponse);
```

Comment:
A couple things:
1. The method `AssertRequestFailsAsync` assumes that the request is a termination request, but you're sending a suspend request.
2. What's the expected behavior for in-proc? Do we throw if you try to suspend a suspended orchestration? I was hoping we wouldn't, but if we are, then checking for a failure message is fine.

Reply:
1. The behavior is the same, but I've updated the parameters and variable names to reflect this. 
2. Yes, we throw in the same way for in-proc (except with a more verbose exception message, as described in the bug linked in the comments)